router/pkg/metric/prom_metric_store.go

```diff
@@ -43,6 +49,18 @@ func NewPromMetricStore(logger *zap.Logger, meterProvider *metric.MeterProvider)
 	return m, nil
 }
 
+func (h *PromMetricStore) DeregisterRegisteredInstruments() error {
```

Comment:
Used this name instead of `Shutdown` as PromMetrics has an existing `Shutdown` method, but does not seem to get called, but erred on the safer side and didn't remove it.